### PR TITLE
build(images): restrict image creation, add correct tags and clean non tagged images

### PR DESCRIPTION
**Pull request by psyray**
Image creation has been restricted to the docker folder
Tags are now generated for the current services, and version (eg: rengine-celery-v2.1.0)
I've also added a manual workflow to remove untagged images

### CODE REVIEW

File: .github/workflows/delete-untagged-images.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      - name: Summary
+        if: always()
+        env:
+          DRY_RUN: ${{ github.event.inputs.dry_run }}
+        run: |
+          echo "## Summary of untagged image deletion" >> $GITHUB_STEP_SUMMARY
+          echo "- Dry run: $DRY_RUN" >> $GITHUB_STEP_SUMMARY
+          echo "- Owner: $OWNER" >> $GITHUB_STEP_SUMMARY
+          echo "- Project: $PROJECT" >> $GITHUB_STEP_SUMMARY
+          echo "Check the logs above for more details on deleted images or images that would have been deleted in dry run mode." >> $GITHUB_STEP_SUMMARY
```
Comment (AnonymousWP) [inline .github/workflows/delete-untagged-images.yml:51]: Please never end a file with no extra blank line.

**Review by AnonymousWP — CHANGES_REQUESTED** [review]
Looks good, minor request though. Please do not forget this in new PRs. You'd say it's not important, but it can actually break scripts etc.

File: .github/workflows/delete-untagged-images.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      - name: Delete untagged images
+        uses: Chizkiyahu/delete-untagged-ghcr-action@v3
+        with:
+          token: ${{ secrets.GHCR_PAT }}
+          repository_owner: ${{ env.OWNER }}
+          repository: ${{ env.PROJECT }}
+          untagged_only: true
+          owner_type: org
+          except_untagged_multiplatform: true
+          dry_run: ${{ github.event.inputs.dry_run }}
```
Comment (AnonymousWP) [inline .github/workflows/delete-untagged-images.yml:40]: This misses a condition of _when_ it should automatically run. Maybe when every release merges to master, remove untagged images.

**Review by AnonymousWP — CHANGES_REQUESTED** [review]

File: .github/workflows/delete-untagged-images.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    steps:
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ vars.GHCR_USERNAME }}
+          password: ${{ secrets.GHCR_PAT }}
+
+      - name: Delete untagged images
+        uses: Chizkiyahu/delete-untagged-ghcr-action@v3
```
Comment (AnonymousWP) [inline .github/workflows/delete-untagged-images.yml:32]:
```suggestion
        uses: Chizkiyahu/delete-untagged-ghcr-action@v4
```
Use latest version: https://github.com/marketplace/actions/delete-untagged-ghcr

File: .github/workflows/delete-untagged-images.yml (hunk 1 above)
Comment (psyray) [reply .github/workflows/delete-untagged-images.yml]: Done

File: .github/workflows/delete-untagged-images.yml (hunk 2 above)
Comment (psyray) [reply .github/workflows/delete-untagged-images.yml]: Added every 2 weeks

File: .github/workflows/delete-untagged-images.yml (hunk 3 above)
Comment (psyray) [reply .github/workflows/delete-untagged-images.yml]: upgraded

**Review by AnonymousWP — APPROVED** [review]

**Pull request merged**